### ISSUE TITLE
Booleans not narrowed in comparisons

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1063]
**Describe the bug**
Booleans in unions of booleans and ints and floats are not narrowed in value comparisons. 

**To Reproduce**
Type check the snippet below.

**Expected behavior**
`bool` is a subclass of `int` and should be narrowed where the other operand is an `int` or `float`.

**Screenshots or Code**
```python
# pyright: strict

from typing import Literal, Union

def false_or_a_float() -> Union[Literal[False], float]:
    ...
foo = false_or_a_float()

reveal_type(foo)
if foo:
    reveal_type(foo)
if foo > 0:
    reveal_type(foo)
if foo <= 0:
    reveal_type(foo)
if foo == 0:
    reveal_type(foo)
# etc.

```

**VS Code extension or command-line**
Extension, 1.174

**Comment by erictraut** [comment]
Pyright doesn't do narrowing of any int based on comparisons, so this specific to bool. For example, if you replace `False` with `0`, you will see the same results.

I question whether this is a common enough idiom to justify the special-case narrowing logic.

**Comment by layday** [comment]
Perhaps not; I noticed mypy was responding to it (it reported a 'superfluous cast' in `foo > 0 and do_something_with_int_returning_bool(cast(int, foo))`) though it is not able to narrow ints in e.g. `0 == 0`.
